cromshell

```diff
@@ -321,6 +329,26 @@ function invalidSubCommand()
   exit 3;
 }
 
+function getAnswerFromUser()
+{
+  local prompt="${1}" 
+  local acceptableValues="${2}"
+  local responseVar="${3}" 
+
+  local haveGoodValue=false
+  while ! $haveGoodValue ; do
+    read -p "${prompt} [$( echo ${acceptableValues} | sed 's# #/#g' )] : " ${responseVar}
+
+    for okVal in ${acceptableValues} ; do
+      if [[ "${!responseVar}" == "${okVal}" ]] ; then
+        haveGoodValue=true
+      fi
+    done
+
+    ! $haveGoodValue && error "Please enter one of the following (CASE SENSITIVE): $( echo ${acceptableValues} | sed 's# #/#g' )" && error ""
```

Comment:
lets make this tr for consistency

Reply:
Yup.  Fixed.

Reply:
This is a funny message because you don't have a chance to re-enter it.  It should either be something like `Error: one of the following was required` or it should loop to let you try again.

Reply:
It does loop around.  The `while` loop only exits if it validates the input from the user.

The `for` loop is doing an exact strict match on allowable values.

Reply:
woops, didn't read carefully enough, sorry